### PR TITLE
Fix slic float32 support regression

### LINKED ISSUE #4682: slic float32 support regression
## Description

As reported by @jklaise in https://github.com/scikit-image/scikit-image/issues/4675#issuecomment-626794531, `slic` no more supports `float32`.

## Way to reproduce
```python
from skimage.segmentation import slic
import numpy as np
im = np.random.rand(28, 28).astype(np.float32)
slic(im)
```

## Version information

```python
3.8.2 (default, Mar 26 2020, 15:53:00) 
[GCC 7.3.0]
Linux-4.4.0-178-generic-x86_64-with-glibc2.10
scikit-image version: 0.16.2
numpy version: 1.18.1
```

### PR DESCRIPTION
**Pull request by rfezzani**
## Description

Fixes #4682 

The use of `np_floats` fused type was already introduced in #4339, but the support for `float32` was not tested :confused: 

`foat32` support is added in this PR and a test for dtype support is added.

## Checklist

- [ ] [Docstrings for all functions](https://github.com/numpy/numpy/blob/master/doc/example.py)
- [ ] Gallery example in `./doc/examples` (new features only)
- [ ] Benchmark in `./benchmarks`, if your changes aren't covered by an
  existing benchmark
- [ ] Unit tests
- [ ] Clean style in [the spirit of PEP8](https://www.python.org/dev/peps/pep-0008/)

## For reviewers

- [ ] Check that the PR title is short, concise, and will make sense 1 year
  later.
- [ ] Check that new functions are imported in corresponding `__init__.py`.
- [ ] Check that new features, API changes, and deprecations are mentioned in
      `doc/release/release_dev.rst`.

### CODE REVIEW

File: skimage/segmentation/tests/test_slic.py [hunk 1]
```diff
@@ -463,3 +464,10 @@ def test_gray_3d_mask():
     assert_equal(len(np.unique(seg)), 9)
     for s, c in zip(slices, range(1, 9)):
         assert_equal(seg[s][2:-2, 2:-2, 2:-2], c)
+
+
+@pytest.mark.parametrize("dtype", ['float32', 'float64', 'uint8', 'int'])
+def test_dtype_support(dtype):
+    img = np.random.rand(28, 28).astype(dtype)
+
+    assert slic(img).dtype == int
```
Comment (hmaarrfk) [inline skimage/segmentation/tests/test_slic.py:473]:
```suggestion
    assert slic(img).dtype == np.dtype(int)
```

File: skimage/segmentation/slic_superpixels.py [hunk 2]
```diff
@@ -272,14 +272,14 @@ def slic(image, n_segments=100, compactness=10., max_iter=10, sigma=0,
...
         [centroids, np.zeros((n_centroids, image.shape[3]))],
-        axis=-1))
+        axis=-1), dtype=dtype)
 
     # Scaling of ratio in the same way as in the SLIC paper so the
     # values have the same meaning
     step = max(steps)
     ratio = 1.0 / compactness
 
-    image = np.ascontiguousarray(image * ratio, dtype=np.double)
```
Comment (alexdesiqueira) [inline skimage/segmentation/slic_superpixels.py:282]: @rfezzani a question: shouldn't this `dtype` continue here as `dtype=dtype`? Or else, if I understood correctly, `image` will return with `dtype=None` (from [np.ascontiguousarray()](https://numpy.org/doc/stable/reference/generated/numpy.ascontiguousarray.html) default `dtype` value) . Is that right? Thanks for clarifying!

Comment (rfezzani) [reply skimage/segmentation/slic_superpixels.py:282]: That's absolutely right @alexdesiqueira, but I will set dtype explicit for caution ;)

Comment (rfezzani) [reply skimage/segmentation/slic_superpixels.py:282]: Done ;)

**Comment by rfezzani** [comment]
Thank you @hmaarrfk ;)

**Comment by emmanuelle** [comment]
Note to self: to be backported to 0.17.x

**Comment by emmanuelle** [comment]
@meeseeksdev backport to v0.17.x

**Comment by rfezzani** [comment]
Thank you @emmanuelle

**Comment by alexdesiqueira** [comment]
Thank you Riadh!

**Pull request merged**